src/components/data/converters/permission.js

```diff
@@ -0,0 +1,12 @@
+/* eslint-disable import/prefer-default-export */
+
+export function getPermissionLabelString(permission, formatMessage, showRaw) {
```

Comment:
In an otherwise elegant solution, it's an unsightly wart that this function needs to have `formatMessage` passed into it — apart from anything else, it requires us to `injectIntl` into the components that use it just so they can obtain the `formatMessage` function. Is there really no way for this function to find that for itself?

Reply:
Unfortunately...no. The issue here is that we eventually need to return a plain JS string. The only way to get that out of react-intl is via `formatMessage`. react-intl only makes _that_ available via React-y methods because all of its translations, locale-data, etc is stored that way and needs to be...Reactive.

Because everything is only available via React-y methods, there's literally no way to get a translated string from a basic execution context like that function. We'd need to get at react-intl first via something like its provided components or injectIntl or useIntl. But then we're in a React component, and we can't return a plain string from that, we can only return a React element. Sigh!

I could've avoided passing formatMessage, but then what happens is that you move some parts of the formatting (eg, the `showRaw` ternary) out of that function. So that's not great either. :\

Reply:
I really do hate that I have to pass in formatMessage so would _definitely_ appreciate a better solution. I just have never found one. I actually ran into this same issue a while ago in ERM code and poked around for a while without finding anything better. :(

Reply:
Also, it should be noted that unless a component _absolutely, positively_ wants a `String` copy of the permission label, it doesn't need to injectIntl/useIntl. If it just wants to render the permission label, it should just render the `PermissionLabel` component.

Reply:
Yes, all of that sounds right. It's mildly disappointing that the way the underlying APIs are set up doesn't allow us to do this more elegantly, but given the lack of alternatives I don't think we need to agonise too much.